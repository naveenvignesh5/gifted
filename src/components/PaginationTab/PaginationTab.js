import React, { memo, useState } from "react";

import "./PaginationTab.sass";
import { DEFAULT_PAGE_COUNT_PER_GROUP } from "../../constants";

const PaginationTab = memo(
  ({
    totalPages,
    currentPage,
    pageLimit = DEFAULT_PAGE_COUNT_PER_GROUP,
    onPagePress,
    onPrevButtonPress,
    onNextButtonPress
  }) => {
    const [startPage, updateStartPage] = useState(1);

    const onPrevPress = () => {
      if (startPage === 1) return;

      const newStartPage = startPage - pageLimit;

      if (newStartPage > 0) {
        updateStartPage(newStartPage);
        onPrevButtonPress(newStartPage);
      }
    };

    const onNextPress = () => {
      if (startPage === totalPages) return;

      const newStartPage = startPage + pageLimit;

      if (newStartPage <= totalPages) {
        updateStartPage(newStartPage);
        onNextButtonPress(newStartPage);
      }
    };

    const renderPageButtons = () => {
      let buttons = [];

      let limit = startPage + pageLimit - 1;

      for (let i = startPage; i <= limit; i += 1) {
        buttons.push(
          <button
            key={`page_${i}`}
            onClick={() => onPagePress(i)}
            className={`page-button ${currentPage === i ? "is-active" : ""}`}
          >
            {i}
          </button>
        );
      }

      return buttons;
    };

    return (
      <div className="is-hidden-mobile">
        <div className="paginatin-tab">
          <div className="page-buttons-wrapper">
            <button className="page-button" onClick={onPrevPress}>
              Prev
            </button>
            {renderPageButtons()}
            <button className="page-button" onClick={onNextPress}>
              Next
            </button>
          </div>
        </div>
      </div>
    );
  }
);

export default PaginationTab;
