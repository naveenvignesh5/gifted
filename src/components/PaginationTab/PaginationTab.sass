@import './../../styles/theme'
@import './../../styles/helper'
    
.paginatin-tab
    display: flex
    flex-direction: column
    align-items: center
    @include themify($themes)
        background-color: themed('paginationBackground')

.page-buttons-wrapper
    display: inline-flex
    justify-content: center
    padding: 2rem 1rem
    width: 60%

.page-button
    padding: 1rem
    min-width: 3rem
    min-height: 3rem
    font-size: 1rem
    font-weight: bold
    border-radius: 10%
    margin: auto 1rem
    @include themify($themes)
        background-color: themed('pageButtonBackground')
        color: themed('pageButtonColor')

    &:hover
        @include themify($themes)
            background-color: themed('pageButtonHoverBackground')
            color: themed('pageButtonHoverColor')

    &.is-active
        @include themify($themes)
            background-color: themed('pageButtonHoverBackground')
            color: themed('pageButtonHoverColor')
