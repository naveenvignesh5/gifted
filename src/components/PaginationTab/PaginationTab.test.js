import React from "react";
import renderer from "react-test-renderer";

import PaginationTab from "./PaginationTab";

describe("PaginationTab", () => {
  it("snapshot renders", () => {
    const component = renderer.create(
      <PaginationTab totalPages={20} currentPage={1} pageLimit={10} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
