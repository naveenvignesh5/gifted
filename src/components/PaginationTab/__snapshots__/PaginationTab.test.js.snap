// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaginationTab snapshot renders 1`] = `
<div
  className="is-hidden-mobile"
>
  <div
    className="paginatin-tab"
  >
    <div
      className="page-buttons-wrapper"
    >
      <button
        className="page-button"
        onClick={[Function]}
      >
        Prev
      </button>
      <button
        className="page-button is-active"
        onClick={[Function]}
      >
        1
      </button>
      <button
        className="page-button "
        onClick={[Function]}
      >
        2
      </button>
      <button
        className="page-button "
        onClick={[Function]}
      >
        3
      </button>
      <button
        className="page-button "
        onClick={[Function]}
      >
        4
      </button>
      <button
        className="page-button "
        onClick={[Function]}
      >
        5
      </button>
      <button
        className="page-button "
        onClick={[Function]}
      >
        6
      </button>
      <button
        className="page-button "
        onClick={[Function]}
      >
        7
      </button>
      <button
        className="page-button "
        onClick={[Function]}
      >
        8
      </button>
      <button
        className="page-button "
        onClick={[Function]}
      >
        9
      </button>
      <button
        className="page-button "
        onClick={[Function]}
      >
        10
      </button>
      <button
        className="page-button"
        onClick={[Function]}
      >
        Next
      </button>
    </div>
  </div>
</div>
`;
